src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -79,7 +79,7 @@ public function denormalize($data, $class, $format = null, array $context = arra
 
             $dateTimeErrors = \DateTime::class === $class ? \DateTime::getLastErrors() : \DateTimeImmutable::getLastErrors();
 
-            throw new UnexpectedValueException(sprintf(
+            throw new NotNormalizableValueException(sprintf(
```

Comment:
I have thought about that too. The reason I did not add it is consistency. We would only be able to add codes for Symfony's own normalizer. However, it's quite common to add your own normalizers in libraries/projects. As a consequence exception codes would only be useful when they are issued by built-in normalizers. I don't really like that. What do you think?